engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -606,29 +530,20 @@ public void saveComponent(long entityId, Component component) {
      * Implementation
      */
 
-    protected void assignToPool(EntityRef ref, EngineEntityPool pool) {
-        //Todo: job for the sector manager?
-        poolMap.put(ref.getId(), pool);
+    private EngineEntityPool getEntityPool(long id) {
+        EngineEntityPool pool = poolMap.get(id);
+        if (pool == null) {
+                logger.error("Entity {} doesn't have an assigned pool", id);
```

Comment:
I kept this in because it was very helpful to debug when I was having issues with entities not being put into the pools correctly.

With the latest round of fixes this shouldn't actually occur, but I think providing a sensible fallback in case of error is worthwhile.

Querying for the pool of a non-existing id isn't something I thought about; it would currently return the global pool, but should probably check for that and return some sort of null value.

We could introduce an EntityPool.NULL, analogous to the EntityRef.NULL. This would provide sensible defaults in the case of an error (i.e. the pool appears empty for any methods calling it).

I'm not sure about the case of an entity that isn't assigned to a pool. It shouldn't currently be possible to have that situation (I have just realised that I should probably make createEntity() protected, rather than public, to stop this kind of stuff). But I think assigning the stray entity to the global pool in case of error is a safe bet.